src/Containers.jsx

```diff
@@ -590,19 +591,64 @@
+    filterContainers = (containers) => {
+        let filtered = [];
+        filtered = Object.keys(containers).filter(id => !(this.props.filter == "running") || ["running", "restarting"].includes(this.props.containers[id]?.State?.Status));
+
+        const filter_by_text = (lcf, id) => {
+            const container = containers[id];
+            const systemd_unit_match = container.Config?.Labels?.PODMAN_SYSTEMD_UNIT?.toLowerCase().indexOf(lcf) >= 0;
+            const name_match = container.Name.toLowerCase().indexOf(lcf) >= 0;
+            const image_match = container.ImageName.toLowerCase().indexOf(lcf) >= 0;
+
+            if (container.Pod) {
+                const podKey = utils.makeKey(container.uid, container.Pod);
+                const pod = this.props.pods[podKey] || this.props.quadletPods[podKey];
+                const pod_match = pod.Name.toLowerCase().indexOf(lcf) >= 0 || pod.Labels?.PODMAN_SYSTEMD_UNIT?.toLowerCase().indexOf(lcf) >= 0;
+                return name_match || systemd_unit_match || image_match || pod_match;
+            } else {
+                return name_match || systemd_unit_match || image_match;
+            }
+        };
 
-        if (container.Pod) {
-            const pod = this.props.pods[utils.makeKey(container.uid, container.Pod)];
-            const pod_match = pod.Name.toLowerCase().indexOf(lcf) >= 0 || pod.Labels?.PODMAN_SYSTEMD_UNIT?.toLowerCase().indexOf(lcf) >= 0;
-            return name_match || systemd_unit_match || image_match || pod_match;
-        } else {
-            return name_match || systemd_unit_match || image_match;
+        if (this.props.ownerFilter !== "all") {
+            filtered = filtered.filter(id => {
+                if (this.props.ownerFilter === "user")
+                    return containers[id].uid === null;
+                return containers[id].uid === this.props.ownerFilter;
+            });
+        }
+
+        if (this.props.textFilter.length > 0) {
+            const lcf = this.props.textFilter.toLowerCase();
+            filtered = filtered.filter(id => filter_by_text(lcf, id));
         }
+
+        // Remove infra and service containers
+        filtered = filtered.filter(id => !containers[id].IsInfra && !containers[id].IsService);
+
+        const getHealth = id => {
+            const state = containers[id]?.State;
+            return state?.Health?.Status || state?.Healthcheck?.Status;
+        };
+
+        filtered.sort((a, b) => {
+            // Show unhealthy containers first
+            const a_health = getHealth(a);
+            const b_health = getHealth(b);
+            if (a_health !== b_health) {
+                if (a_health === "unhealthy")
+                    return -1;
+                if (b_health === "unhealthy")
+                    return 1;
+            }
+            // User containers are in front of system ones
+            if (containers[a].uid !== containers[b].uid)
+                return (containers[a].uid === 0) ? 1 : -1;
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2266-29c256e1-20250917-140235-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)